src/gpuarray_collectives_cuda_nccl.c

```diff
@@ -175,11 +175,14 @@ static inline ncclDataType_t convert_data_type(int typecode) {
   switch (typecode) {
   case GA_BYTE: return ncclChar;
   case GA_INT: return ncclInt;
-  case GA_HALF: return ncclHalf;
   case GA_FLOAT: return ncclFloat;
   case GA_DOUBLE: return ncclDouble;
   case GA_LONG: return ncclInt64;
   case GA_ULONG: return ncclUint64;
+  #ifdef CUDA_HAS_HALF
```

Comment:
I remember that, but I think GA_HALF is always defined. So this cause a problem if libgpuarray support it while nccl don't. Here we don't give a good error message. Here, we request cuda 7 or more recent:

http://deeplearning.net/software/libgpuarray/installation.html#run-requirements

And CUDA_HAS_HALF is always true in that case. So I would remove the ifdef.